ec/ecresource/trafficfilterresource/flatteners.go

```diff
@@ -18,67 +18,64 @@
 package trafficfilterresource
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 
 	"github.com/elastic/cloud-sdk-go/pkg/models"
 )
 
-func modelToState(d *schema.ResourceData, res *models.TrafficFilterRulesetInfo) error {
-	if err := d.Set("name", *res.Name); err != nil {
-		return err
-	}
-
-	if err := d.Set("region", *res.Region); err != nil {
-		return err
-	}
+func modelToState(ctx context.Context, res *models.TrafficFilterRulesetInfo, state *modelV0) diag.Diagnostics {
+	var diags diag.Diagnostics
 
-	if err := d.Set("type", *res.Type); err != nil {
-		return err
-	}
-
-	if err := d.Set("rule", flattenRules(res.Rules)); err != nil {
-		return err
-	}
+	state.Name = types.String{Value: *res.Name}
+	state.Region = types.String{Value: *res.Region}
+	state.Type = types.String{Value: *res.Type}
+	state.IncludeByDefault = types.Bool{Value: *res.IncludeByDefault}
 
-	if err := d.Set("include_by_default", res.IncludeByDefault); err != nil {
-		return err
-	}
+	diags.Append(flattenRules(ctx, res.Rules, &state.Rule)...)
 
 	if res.Description != "" {
```

Comment:
It looks like `Description.Null` if false here. Shall we remove the `if` check or set `Null` to `true` if `Description` is empty?

Reply:
Fixed.